src/main/resources/public/ts/controllers/eval_export_controller.ts

```diff
@@ -369,32 +368,42 @@ export let exportControleur = ng.controller('ExportController', ['$scope',
         $scope.getYearsAndPeriodes = function () {
             $scope.currentYearTypesPeriodes = [];
             $scope.years = [];
-            http.get('/competences/years?idStructure=' + $scope.structure.id).then(function (data) {
+
+            let url = '/competences/archive/years?idStructure=' + $scope.structure.id +
+                '&type=' + $scope.paramsArchive.type;
+            http.get(url).then(function (data) {
                 if(data.status === 200){
                     let res = data.data;
 
-                    let periodes = JSON.parse(res.periodes);
-                    $scope.currentYearTypesPeriodes = _.filter($scope.structure.typePeriodes.all , (type) => {
-                        type.selected = true;
-                        return periodes.includes(type.id);
-                    });
+                     let active_year = res.active_year;
+                     $scope.currentYearTypesPeriodes = _.filter($scope.structure.typePeriodes.all , (type) => {
+                         type.selected = true;
+                         return active_year.periodes.includes(type.id);
+                     });
 
-                    let startYear = moment(res.start_date).format('YYYY');
-                    let endYear = moment(res.end_date).format('YYYY');
-                    $scope.years.push({
-                        id: startYear,
-                        libelle : startYear + ' - ' + endYear +
-                            " (" + lang.translate('viescolaire.utils.annee.encours') + ")"
-                    });
+                    let years = res.years;
+                    let start_active_year = active_year.start_date.substring(0, 4);
+                    let end_active_year = active_year.end_date.substring(0, 4);
+                    if(years.length > 0) {
```

Comment:
On est sur que `years` sera toujours défini ? Tu peux risquer d'avoir un 'cannot read property length of undefined | null'